### ISSUE TITLE
Handshake failure with RSA-PSS key and reversed "signature_algorithms" order

### ISSUE THREAD

**Issue by ocheron** [openssl/openssl#9545]
It seems the order of signature algorithms received from the client impacts ability for a server to select the hash algorithm satisfying key restrictions.

Steps to reproduce:

1. Generate an RSA-PSS key pair restricted to SHA-256 and start a TLS server:
```shell
$ openssl req -x509 -new -nodes -subj /CN=localhost -newkey rsa-pss -pkeyopt rsa_pss_keygen_md:sha256 -pkeyopt rsa_pss_keygen_saltlen:32 -out cert.pem
$ openssl s_server -key privkey.pem -cert cert.pem
```

2. Try to connect from a client with "signature_algorithms" ordered by decreasing hash size:
```shell
$ openssl s_client -CAfile cert.pem -sigalgs rsa_pss_pss_sha512:rsa_pss_pss_sha384:rsa_pss_pss_sha256
CONNECTED(00000004)
depth=0 CN = localhost
verify return:1
4627883456:error:14094438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl/record/rec_layer_s3.c:1536:SSL alert number 80
...
```

The server fails with:
```
ERROR
4511495616:error:0408F091:rsa routines:pkey_rsa_ctrl:digest not allowed:crypto/rsa/rsa_pmeth.c:506:
4511495616:error:141F0006:SSL routines:tls_construct_cert_verify:EVP lib:ssl/statem/statem_lib.c:273:
```

Connection is successful when giving `-sigalgs rsa_pss_pss_sha256:rsa_pss_pss_sha384:rsa_pss_pss_sha512` instead.

Observed with:
```shell
$ openssl version
OpenSSL 1.1.1c  28 May 2019
```

**Comment by mattcaswell** [comment]
Fix in #9553

**Comment by ocheron** [comment]
Working fine now, thank you.
